.github/scripts/validate_examples.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+"""Validate example_pipelines modules by compiling every exported pipeline."""
+
+from __future__ import annotations
+
+import argparse
+import importlib.util
+import sys
+import tempfile
+import traceback
+from pathlib import Path
+from types import ModuleType
+from typing import Iterable, List, Sequence, Set, Tuple
+
+try:
+    from kfp import compiler, dsl
+except ImportError as exc:  # pragma: no cover - surfaced to callers
+    raise SystemExit(
+        "kfp is required to validate example pipelines. "
+        "Install it with `pip install kfp`."
+    ) from exc
+
+REPO_ROOT = Path(__file__).resolve().parents[2]
+DEFAULT_TARGETS = (REPO_ROOT / "components", REPO_ROOT / "pipelines")
+PIPELINE_ATTRS = ("pipeline_func", "_pipeline_func", "pipeline_spec")
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description=(
+            "Import example_pipelines.py modules for the specified components or pipelines "
+            "and compile every @dsl.pipeline function they export."
+        )
+    )
+    parser.add_argument(
+        "paths",
+        metavar="PATH",
+        nargs="*",
+        help=(
+            "Component or pipeline directories (or files within them). "
+            "If omitted, every example_pipelines.py file is validated."
+        ),
+    )
+    return parser.parse_args()
+
+
+def normalize_targets(raw_paths: Sequence[str]) -> List[Path]:
```

Comment:
duplicate function with the one in run_tests - maybe centralize into a utility library or script (determine_changed_assets, maybe)?